mlir/lib/Dialect/Rock/Generator/ConvGenerator.cpp

```diff
@@ -499,36 +484,35 @@ LogicalResult ConvGenerator::parseConvConfig(OpBuilder &builder,
 
   auto isValid = [&argMap]() {
     // only require tensor configs
-    static const std::vector<std::string> validKeys = {
+    static std::vector<std::string> validKeys = {
         "batchsize",   "groupsize",    "in_layout", "in_type",
         "in_channels", "in_h",         "in_w",      "out_layout",
         "out_type",    "out_channels", "out_h",     "out_w",
         "fil_layout",  "fil_type",     "fil_w",     "fil_h"};
-    if (!std::all_of(validKeys.cbegin(), validKeys.cend(),
-                     [&argMap](const std::string &key) {
-                       return argMap.count(key) > 0;
-                     })) {
-      return false;
+    if (argMap["in_layout"].length() > 5) { // Ie, 3-D.
```

Comment:
Might be nice to have a second vector of keys to `all_of` on for the 3D case but that's me.